### ISSUE TITLE
PHP 7.1 + Pecl-Redis-3.1.1RC2 - ignored session.save_path in php.ini

### ISSUE THREAD

**Issue by janreges** [phpredis/phpredis#1097]
Hi,

i don't know, whether it is issue in PHP-FPM, or in Pecl-Redis library.

My php.ini for PHP-FPM contains:
```
session.save_handler = redis
session.save_path = tcp://127.0.0.1:6379
```
phpinfo() also shows this proper setting.

But PHP scripts that are using session, end up with exception "Connection closed".

I discovered, that when i set `php_admin_value[session.save_path] = tcp://127.0.0.1:6379`  in `php-fpm.conf`, it works.

So, PHP-FPM or Pecl-Redis library ignore `session.save_path` value from `php.ini` (visible also in phpinfo()) and it is required to force it by `php_admin_value` in `php-fpm.conf`.

Thank you for your check.

**Comment by remicollet** [comment]
if session.save_handler is set in pool configuration (which is the case in some linux distro, Fedora, RHEL, Centos...) you indeed have to change it there. This is the expected behavior.

TLDR: having a global configuration in php.ini for session.save_path was terribly wrong, as it raise issue, especially when "root" run some scripts which cannot be clean later by other user (apache, fpm, nginx...). So having a per-SAPI configuration, and so, per FPM pool, fixes this old issue, and is much more flexible, BTW, this should be documented in the php.ini, ex:
```
; RPM note : session directory must be owned by process owner
; for mod_php, see /etc/httpd/conf.d/php.conf
; for php-fpm, see /etc/php-fpm.d/*conf


```

**Comment by janreges** [comment]
Aaa, thank you @remicollet ! 👍 

It makes sense, thank you for your explanation.

Main confusion is, that phpinfo() contains session.save_path with expected value. Requirement to overwriting it from php-fpm.conf to same value is weird, but working solution.

Thank you for your support and have a nice day.
